### ISSUE TITLE
Ошибки большой тестовой коллекции

### ISSUE THREAD

**Issue by konstantin-smith** [dialogue-evaluation/factRuEval-2016#33]
Некоторые тексты имеют сбойные кодировки UTF-8: начинаются нормально, но затем сбой почти до конца, в конце немного нормально.
Например, 757, 1335, 1344, 1805 и др.

**Comment by vbocharov** [comment]
Вижу проблему. У меня получился 181 такой документ. Есть ли кто-то, у кого больше документов с проблемами в UTF-8?

Update: Нашёл ещё. 276 всего

**Comment by vbocharov** [comment]
Исправил. Отдал в тестирование. Днём разошлём всем.

**Comment by dsoloviev** [comment]
Для обучения данные только то, что в devset?

**Comment by vbocharov** [comment]
@dsoloviev да

**Comment by FedorBykov** [comment]
Коллеги, объясните, пожалуйста, о какой коллекции идет речь? Тестовая коллекция уже появилась?

**Comment by vbocharov** [comment]
@FedorBykov да. Ссылку присылает оргкомитет Диалога после заполнения вот этой формы:  http://goo.gl/forms/DLqEVW4NLf . Об этом было вчера письмо на рассылку.

**Comment by konstantin-smith** [comment]
Для регистрация результатов прогона участник должен прислать оргкомитету
через Googleформу    ● ссылку на архив с результатами прогона ...

А где эта форма?

**Comment by asolov** [comment]
а зачем столько пустых файлов?
book_10220.txt
book_13586.txt
book_16874.txt
book_16952.txt
book_17957.txt
book_18031.txt
book_19446.txt
book_20105.txt
book_20803.txt
book_21194.txt
book_22832.txt
book_24454.txt
book_2448.txt
book_27857.txt
book_28984.txt
book_32277.txt
book_33403.txt
book_34373.txt
book_35771.txt
book_7560.txt
book_8529.txt

**Comment by vbocharov** [comment]
@konstantin-smith 
вот: http://goo.gl/forms/snXO6MeHDo

**Comment by vbocharov** [comment]
@asolov игнорируйте их. Эти статьи раньше были в вебе, но в этот обход там оказалась заглушка - получились пустые строки.

**Comment by konstantin-smith** [comment]
@vbocharov 
Ещё уточнение насчёт результата. Я правильно понял, что в архиве должны быть только .task1 \ .task2 \ .task3 - файлы?

**Comment by vbocharov** [comment]
@konstantin-smith да

**Comment by asolov** [comment]
Виктор, хотелось бы уточнить: четырехбайтные символы считать как два символа или как один (отображаются как один, занимают место как два)?

**Comment by vbocharov** [comment]
@asolov Если я правильно понял вопрос, то речь идёт о UTF-8 символах длиной 4 байта. Если так, то да, это один символ, не важно, какой он длины.

**Comment by asolov** [comment]
тогда совсем не понятно... вот я нашел в описании:
- Все смещения и длины считаются в символах Unicode, а не в байтах. 
  4 байтные символы занимают два юникод-символа в utf8 , но графически отображаются как один.
  т.е. получается нужно смещаться на два символа?

**Comment by sysoev-a** [comment]
@asolov Прошу прощения, что вмешиваюсь, но Вы не про суррогаты спрашиваете? Я в тестовой коллекции таких не находил.
Может быть, Вы укажете, где конкретно возникает проблема?

**Comment by asolov** [comment]
есть большое кол-во 4-байтных смайлов, которые отображаются как один символ, а занимают два. Конкретно в коллекции много списков, типа:
• как пожелание , заполненная таблица должна образовывать
- первый трехбайтовый символ в ютф8 занимает две позиции, но отображается как один.

**Comment by sysoev-a** [comment]
Судя по всему, Вы говорите вот об этом символе: http://www.fileformat.info/info/unicode/char/2022/index.htm
Да, в UTF-8 он занимает 3 байта. И да, он отображается как один символ. Но я не могу понять, почему вы считаете, что он "занимает две позиции".

**Comment by asolov** [comment]
да, действительно один, тут я ошибся. 4_байтных пока не нашел, так что вопрос снимается.

**Comment by alexandr-j** [comment]
Добрый день,
Немного критики тестовой коллекции.
В регламенте соревнования сказано, что "Корпус текстов соревнования состоит из новостных и аналитических текстов на общественнополитические темы". Тем не менее легко видеть, что это очень часто не так:
book__20 - статья  №56 из какого-то кодекса
book_17828 - текст, про то, кто такие весталки
book_5860 - о том, что такое станок-качалка
book_7080 - Основы системного проектирования на 4000 строк.
book_11899 - статья "Осип Мандельштам в революционной Москве" с отрывками стихотворений. Ну ладно, это пусть будет "общественно-политическое", хотя бедный Мандельштам.

И таких документов чуть ли не большинство. У меня было похожее ощущение, когда я прогонял новостной анализатор по Войне и Миру. 
Ну и статистика: в среднем в русскоязычном новостном потоке (я говорю о СМИ, а не о блогах или соцсетях) - примерно каждый 20-й документ содержит слово "ООО". А в тестовой коллекции - каждый 110-й. То есть выборка не слишком пригодна для тренировки обработки реального новостного потока. 
Ничего формально плохого во всем этом нет, но не понимаю, почему такой выбор текстов сделан для данной задачи.

**Comment by kzn** [comment]
Добрый день.

В тестовой коллекции есть повторяющиеся тексты. Так и задумано?

~/testset% md5sum *.txt | cut -f1 -d ' '|sort |wc -l
37217
~/testset% md5sum *.txt | cut -f1 -d ' '|sort | uniq | wc -l
36789
